splice/web/api/content_upload.py

```diff
@@ -0,0 +1,209 @@
+import hashlib
+import os
+import base64
+import json
+import requests
+import zipfile
+import tempfile
+
+from furl import furl
+from mohawk import Sender
+from boto.s3.key import Key
+from splice.web.api.s3_common import setup_s3, MIME_EXTENSIONS
+from splice.environment import Environment
+
+
+env = Environment.instance()
+_ORIGINAL_NAME = "original.zip"
+
+
+def _get_original_content(name, version, bucket):
+    original = tempfile.NamedTemporaryFile(prefix='splice', suffix='.zip')
+    s3_key = "{0}/v{1}/{2}".format(name, version, _ORIGINAL_NAME)
+    k = Key(bucket)
+    k.key = s3_key
+    try:
+        k.get_file(original)
+    except:
+        original.close()
+        return None
+    else:
+        return original
+
+
+def resign_content(name, version):
+    """Re-sign a content till the given version. It downloads the original
+    content file from S3, and only re-signs the assets in the manifest. Also,
+    it won't change the version of the content.
```

Comment:
Hmm, it's kind of confusing here.

Should we consider re-signing the content in-place for all the existing version? Say we've already had 3 different versions for a content. They're all being used by Ffx users. If the signing key gets compromised. Shouldn't we just re-sign those 3 in-place?